journalbeat/docs/config-options.asciidoc

```diff
@@ -0,0 +1,124 @@
+the +{beatname_lc}.inputs+ section of the +{beatname_lc}.yml+ file. Each path
+can be a directory path (to collect events from all journals in a directory), or
+a file path. For example:
+
+["source","sh",subs="attributes"]
+----
+{beatname_lc}.inputs:
+- paths:
+  - "/dev/log"
+  - "/var/log/messages/my-journal-file.journal"
+----
+
+Within the configuration file, you can also specify options that control how
+{beatname_uc} reads the journal files and which fields are sent to the
+configured output. See <<{beatname_lc}-options>> for a list of available
+options.
+
+The following examples show how to configure {beatname_uc} for some common use
+cases.
+
+[[monitor-multiple-journals]]
+.Example 1: Monitor multiple journals under the same directory
+This example configures {beatname_uc} to read from multiple journals that
+are stored under the same directory. {beatname_uc} merges all journals under the
+directory into a single journal and reads them. With `seek` set to `cursor`,
+{beatname_uc} starts reading at the beginning of the journal, but will continue
+reading where it left off after a reload or restart.
+["source","sh",subs="attributes"]
+----
+{beatname_lc}.inputs:
+- paths: ["/path/to/journal/directory"]
+  seek: cursor
+----
+
+[[filter-using-field-names]]
+.Example 2: Fetch log events for Redis running on Docker (uses field names from systemd)
+This example configures {beatname_uc} to fetch log events for Redis running in a
+Docker container. The fields are matched using field names from the systemd
+journal.
+["source","sh",subs="attributes"]
+----
+{beatname_lc}.inputs:
+- paths: []
+  include_matches:
+    - "CONTAINER_TAG=redis"
+    - "_COMM=redis"
+----
+
+[[filter-using-translated-names]]
+.Example 3: Fetch log events for Redis running on Docker (uses translated field names)
+This example also configures {beatname_uc} to fetch log events for Redis running in a
+Docker container. However, in this example the fields are matched using the
+translated field names provided by {beatname_uc}.
+["source","sh",subs="attributes"]
+----
+{beatname_lc}.inputs:
+- paths: []
+  include_matches:
+    - "container.image.tag=redis"
+    - "process.name=redis"
+----
+
+[id="{beatname_lc}-options"]
+[float]
+=== Configuration options
+You can specify the following options to configure how {beatname_uc} reads the
+journal files.
+
+[float]
+[id="{beatname_lc}-paths"]
+==== `paths`
+
+A list of paths that will be crawled and fetched. Each path can be a directory
+path (to collect events from all journals in a directory), or a file path. If
+you specify a directory, {beatname_uc} merges all journals under the directory
+into a single journal and reads them. 
+
+[float]
+[id="{beatname_lc}-seek"]
+==== `seek`
+
+The position to start reading the journal from. Valid settings are:
+
+* `head`: Starts reading at the beginning of the file.
+* `tail`: Starts reading at the end of the file.
```

Comment:
what happens on journalbeat restart?

Reply:
Starts from head or tail depending on the configuration. State info from the registry is overwritten.

Reply:
@kvch 
> State info from the registry is overwritten.

Do users need to know this detail?  Feels like implementation. Is it good enough to describe behavior based on the setting specified?

Reply:
The registry overwrite will change where the 'cursor' setting starts upon restart.

My questions was more about: what happens if there are logs and the beat is restarted.

From testing:

`head`: always resends all logs available in the journal

`tail`: always resends the last log -> If no log has been written journalbeat will send the last send log again (restart can lead to duplicates). If multiple log messages have been written to a journal while journalbeat was down, then these logs will not be send (only the very last log message). @kvch I wonder if we want to ignore the last message when doing 'tail' and only collect new ones (always).

Reply:
Let's ignore the last one in case of `tail`.